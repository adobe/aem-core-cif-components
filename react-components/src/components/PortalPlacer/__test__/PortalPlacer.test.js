/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ~ Copyright 2021 Adobe
 ~
 ~ Licensed under the Apache License, Version 2.0 (the "License");
 ~ you may not use this file except in compliance with the License.
 ~ You may obtain a copy of the License at
 ~
 ~     http://www.apache.org/licenses/LICENSE-2.0
 ~
 ~ Unless required by applicable law or agreed to in writing, software
 ~ distributed under the License is distributed on an "AS IS" BASIS,
 ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 ~ See the License for the specific language governing permissions and
 ~ limitations under the License.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
import React from 'react';

const { render } = require('@testing-library/react');

import PortalPlacer from '../PortalPlacer';

describe('PortalPlacer', () => {
    beforeEach(() => {
        document.body.innerHTML = '';
    });

    const MockComponent = props => {
        return (
            <div>
                <span>MockComponent</span>
                <pre>{JSON.stringify(props, null, 4)}</pre>
            </div>
        );
    };

    it('adds a given component to each element of the given selector', () => {
        // Add targets
        [{ a: 'test', c: 456 }, { b: '' }].map(dataset => {
            const target = document.createElement('div');
            target.className = 'target';
            for (const [key, value] of Object.entries(dataset)) {
                target.dataset[key] = value;
            }
            document.body.appendChild(target);
            return target;
        });

        // Add MockComponent to each target
        render(<PortalPlacer selector=".target" component={MockComponent} />);

        expect(document.body).toMatchSnapshot();
    });

    it('renders empty for an invalid selector', () => {
        render(<PortalPlacer selector=".target" component={MockComponent} />);

        expect(document.body).toMatchSnapshot();
    });
});
