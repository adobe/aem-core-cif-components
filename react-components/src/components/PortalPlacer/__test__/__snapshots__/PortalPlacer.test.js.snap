// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortalPlacer adds a given component to each element of the given selector 1`] = `
<body>
  <div
    class="target"
    data-a="test"
    data-c="456"
  >
    <div>
      <span>
        MockComponent
      </span>
      <pre>
        {
    "a": "test",
    "c": "456"
}
      </pre>
    </div>
  </div>
  <div
    class="target"
    data-b=""
  >
    <div>
      <span>
        MockComponent
      </span>
      <pre>
        {
    "b": true
}
      </pre>
    </div>
  </div>
  <div />
</body>
`;

exports[`PortalPlacer renders empty for an invalid selector 1`] = `
<body>
  <div />
</body>
`;
